src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala

```diff
@@ -41,10 +42,14 @@ package object metadata extends DebugEnhancedLogging {
   val allowedFilesXmlNamespaces = List(dcNamespace, dctermsNamespace)
   val allowedAccessRights = List("ANONYMOUS", "RESTRICTED_REQUEST", "NONE")
 
-  val doiPattern: Regex = raw"^10(\.\d+)+/.+".r
+  val doiPattern: Regex = "^10(\\.\\d+)+/.+".r
+  val doiUrlPattern: Regex = "^((https?://(dx\\.)?)?doi\\.org/(urn:)?(doi:)?)?10(\\.\\d+)+/.+".r
+  val urnPattern: Regex = "^urn:[A-Za-z0-9][A-Za-z0-9-]{0,31}:[a-z0-9()+,\\-\\\\.:=@;$_!*'%/?#]+$".r
```

Comment:
Not sure but would triple quotes require half of the back slashes?

Shouldn't the last group also match `A-Z``

A technique to make it more readable:

```suggestion
  val urnPattern: Regex = {
     val an = "A-Za-z0-9"
     s"^urn:[$an][$an-]{0,31}:[$an()+,\\-\\\\.:=@;$_!*'%/?#]+$".r
   }
```

Perhaps check the ASCII table to reduce a bunch of loose characters to the range `(-;`. Don't know how it reacts when overlapping with `$an`

Reply:
> Not sure but would triple quotes require half of the back slashes?

Yes they do! Great suggestion!

Reply:
> Shouldn't the last group also match `A-Z`

Don't know; not sure where the regex is coming from. @lindareijnhoudt do you know?

Reply:
> A technique to make it more readable

That would make the regex not being recognized as such by IntelliJ/Scala; it will no longer try to check the syntax 😢